crates/ra_syntax/src/lib.rs

```diff
@@ -168,6 +168,41 @@ impl SourceFile {
     }
 }
 
+impl crate::ast::Path {
```

Comment:
Let's use `impl ast::Path` here and elsewhere. In retrospect, `pub use crate::ast::SourceFile;` was a bad idea, but that should be fixed in a separate PR. 

Otherwise, LGTM!

bors d+